From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_linux@m4x.org>
Date: Sat, 18 Nov 2017 18:58:48 +0100
Subject: [PATCH] {BUG in 4.15-rc1} drivers/media/dvb-core: initialize err

clang complains:

    drivers/media/dvb-core/dvb_frontend.c:2425:7: error: variable 'err'
    is used uninitialized whenever 'if' condition is false
    [-Werror,-Wsometimes-uninitialized]
                    if (fe->ops.read_ucblocks) {
                        ^~~~~~~~~~~~~~~~~~~~~
    drivers/media/dvb-core/dvb_frontend.c:2466:9: note: uninitialized
    use occurs here
            return err;
                   ^~~
    drivers/media/dvb-core/dvb_frontend.c:2425:3: note: remove the 'if'
    if its condition is always true
                    if (fe->ops.read_ucblocks) {
                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~
---
 drivers/media/dvb-core/dvb_frontend.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/media/dvb-core/dvb_frontend.c b/drivers/media/dvb-core/dvb_frontend.c
index 3ad83359098b..9eff8ce60d53 100644
--- a/drivers/media/dvb-core/dvb_frontend.c
+++ b/drivers/media/dvb-core/dvb_frontend.c
@@ -2107,7 +2107,7 @@ static int dvb_frontend_handle_ioctl(struct file *file,
 	struct dvb_frontend *fe = dvbdev->priv;
 	struct dvb_frontend_private *fepriv = fe->frontend_priv;
 	struct dtv_frontend_properties *c = &fe->dtv_property_cache;
-	int i, err;
+	int i, err = -ENOTSUPP;
 
 	dev_dbg(fe->dvb->device, "%s:\n", __func__);
 
-- 
